src/FitDiagnostics.cc

```diff
@@ -787,15 +794,17 @@ void FitDiagnostics::getNormalizations(RooAbsPdf *pdf, const RooArgSet &obs, Roo
         for (IH h = sigByCh.begin(), eh = sigByCh.end(); h != eh; ++h) sigByCh1[h->first] = (TH1*) h->second->Clone();
         for (IH h = bkgByCh.begin(), eh = bkgByCh.end(); h != eh; ++h) bkgByCh1[h->first] = (TH1*) h->second->Clone();
         for (int t = 0; t < ntoys; ++t) {
-            // zero out partial sums
-            for (IH h = totByCh1.begin(), eh = totByCh1.end(); h != eh; ++h) h->second->Reset();
-            for (IH h = sigByCh1.begin(), eh = sigByCh1.end(); h != eh; ++h) h->second->Reset();
-            for (IH h = bkgByCh1.begin(), eh = bkgByCh1.end(); h != eh; ++h) h->second->Reset();
+	    // zero out partial sums
+	  for (IH h = totByCh1.begin(), eh = totByCh1.end(); h != eh; ++h) h->second->Reset(), diff_tot[h->first] = 0.0;
+	  for (IH h = sigByCh1.begin(), eh = sigByCh1.end(); h != eh; ++h) h->second->Reset(), diff_sig[h->first] = 0.0;
+	  for (IH h = bkgByCh1.begin(), eh = bkgByCh1.end(); h != eh; ++h) h->second->Reset(), diff_bkg[h->first] = 0.0;
```

Comment:
These lines don't do anything when not running with `--saveShapes`, so the uncertainties are nonsense if you run with `--saveNormalizations --saveWithUncertainties` (they are ok with `--saveNormalizations --saveShapes --saveWithUncertainties`, but we should be able to save the correct total norms with uncertainties regardless of whether the shapes are produced :-) ). I think it would be better to save a vector of the channel names in the loop that starts at L671, and then loop through that vector here, setting the different `diff_...[channel]` values to 0.

Reply:
@dittmer What is the timescale for addressing this? It would be great if we could merge this nice feature soon :-)